### PR TITLE
ci: Serve xDSL sources instead of built wheel

### PR DESCRIPTION
**Pull request by PapyChacal**
Follow-up to #304 : Make the same first cells of the notebooks work correctly for local, development usage and online documentation usage.

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
This looks good to me. Let's see what others think.

File: docs/irdl.ipynb [hunk 1]
```diff
@@ -22,6 +22,46 @@
   {
    "cell_type": "code",
    "execution_count": 1,
+   "id": "1c670f99",
```
Comment (georgebisbas) [inline docs/irdl.ipynb:25]: This is sth we definitely do not want

**Review by georgebisbas — CHANGES_REQUESTED** [review]

File: docs/irdl.ipynb (hunk 1 above)
Comment (superlopuh) [reply docs/irdl.ipynb:25]: What don't we want and why?

Comment (math-fehr) [reply docs/irdl.ipynb:25]: I think he means that we want to remove the output of a cell.
We prefer cleaning the outputs first before pushing it!

File: docs/irdl.ipynb [hunk 2]
```diff
@@ -22,6 +22,46 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "7ac54610",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add the local xdsl definition to the path\n",
```
Comment (math-fehr) [inline docs/irdl.ipynb:57]: So if I understand well, this works both in Pyodide and a local Jupyter?

**Review by math-fehr — APPROVED** [review]
I did not follow all the changes here, but I trust you on that!
I would just remove the notebook outputs first though!

File: docs/irdl.ipynb (hunk 2 above)
Comment (PapyChacal) [reply docs/irdl.ipynb:57]: Indeed!
It followed from a discussion with @superlopuh who likes to use the notebooks to check things incrementally during development.
Using that allows to import directly from the sources, so the local code changes take effect.
Maybe an installed xDSL would shadow it though, I'm not sure about that.

File: docs/irdl.ipynb (hunk 1 above)
Comment (georgebisbas) [reply docs/irdl.ipynb:25]: The output should not be there, an din my opinion, neither the installation.
It doesn't look like a good practice.
Have you seen any other projects doing sth similar?

**Review by georgebisbas — CHANGES_REQUESTED** [review]
My opinion is totally in favour of NOT installing xdsl within the notebook.
What if someone has already a different version, or overlapping requirements?
What python path is used?
The most important is that this PR moves CI infrastructure from the CI-code to the core-repo code, which ic just bad IMHO.

File: docs/irdl.ipynb (hunk 1 above)
Comment (PapyChacal) [reply docs/irdl.ipynb:25]: Got it, I'll clean the outputs.
I have little experience in notebooks so did not see this yet, but I know some do in JupyterLite context, as it is a limitation of its Python kernel Pyodide.
Pyodide _has to_ install packages at runtime, and for every restart. Currently the only way around that is to rebuild a whole Pyodide kernel with our package included, which is not yet easily automated. They are thinking about more straightforward ways, so I was thinking about waiting for those.

There may be some way to hide the pip install command from the user, but still run it at kernel startup time. Would that option satisfy you?

**Comment by superlopuh** [comment]
I agree! This is why I thought we might want to discuss this option, which avoids the xdsl install. I think we should probably have two distinct sets of notebooks, one where the expectation is that the user pip installs xdsl, and another where the expectation is that they have xdsl cloned on their machine.

This solution works better for hacking on the local xdsl repo at the same time as the notebook, and the "do nothing" is better if the assumption is that the user calls `pip install xdsl`. Writing out this comment just now I think that we probably want to optimise for the pip install xdsl use case, so I don't think these changes are necessarily what we want for these notebooks.

I'll rework the PR to include commented out code to add to the path for people who do happen to clone it.

**Comment by tobiasgrosser** [comment]
I guess we can close this one @PapyChacal ?

**Comment by PapyChacal** [comment]
Ah yes, clearly, thanks for telling!
Fore reference: closing this PR for a better solution in #328

**Pull request closed without merging**